### PR TITLE
Fixes check in `StreamLinifier` for pending lines

### LINKED ISSUE #430: TestInteractiveREPL is flaky
```
--- FAIL: TestSQLREPLTestSuite (14.27s)
    --- FAIL: TestSQLREPLTestSuite/TestInteractiveREPL (6.16s)
        require.go:44: 
            	Error Trace:	sql_test.go:89
            	Error:      	"show tables;
            	            	
            	            	+--------------+
            	            	| Table        |
            	            	+--------------+
            	            	
            	            	| blobs        |
            	            	| commit_blobs |
            	            	| commit_files |
            	            	| commit_trees |
            	            	| commits      |
            	            	| files        |
            	            	| ref_commits  |
            	            	| refs         |
            	            	| remotes      |
            	            	| repositories |
            	            	| tree_entries |
            	            	+--------------+
            	            	" does not contain "+--------------+
            	            	| Table        |
            	            	+--------------+
            	            	| blobs        |
            	            	| commit_blobs |
            	            	| commit_files |
            	            	| commit_trees |
            	            	| commits      |
            	            	| files        |
            	            	| ref_commits  |
            	            	| refs         |
            	            	| remotes      |
            	            	| repositories |
            	            	| tree_entries |
            	            	+--------------+
            	            	"
            	Test:       	TestSQLREPLTestSuite/TestInteractiveREPL
```

most probably there is some problem with smart line parser.

### PR DESCRIPTION
**Pull request by se7entyse7en**
Closes #430.

### CODE REVIEW

**Comment by smacker** [comment]
Any chance for a test?

**Comment by se7entyse7en** [comment]
@smacker Added tests.

BTW `StreamLinifier` is currently in `common.go` that has the `// +build integration regression` build tag. So I've also added `// +build integration` to `common_test.go` (maybe we should add also `regression`).  But `common_test.go` doesn't actually contain any integration test, but tests for helpers used for integration tests. Do you think it is better to move it to another place and remove the build tag?

File: cmdtests/common_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+		close(in)
+	}()
+
+	out := sl.Linify(in)
+	counter := 1
+	for m := range out {
+		require.Equal(m, fmt.Sprintf("line %d", counter))
+		counter++
+	}
```
Comment (smacker) [inline cmdtests/common_test.go]: looks like you missed the check for number on produced lines (which must be equal to `messagesNum`)

File: cmdtests/common_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+func (s *StreamLinifierTestSuite) TestOneLineOnMultieMessages() {
```
Comment (smacker) [inline cmdtests/common_test.go]: is here a typo?

Comment (se7entyse7en) [reply cmdtests/common_test.go]: Fixed [here](https://github.com/src-d/engine/pull/432/commits/e897403321eafdbf478b540f898d8386342d7b75).

File: cmdtests/common_test.go (hunk 1 above)
Comment (se7entyse7en) [reply cmdtests/common_test.go]: Thanks! Addressed [here](https://github.com/src-d/engine/pull/432/commits/3ce3ebb3d844900e3233f25efac76aa7600ca268).

**Comment by se7entyse7en** [comment]
Rebased and squashed. Gonna merge when all green.

**Pull request merged**